shared/tree-sitter-extractor/src/extractor/mod.rs

```diff
@@ -4,12 +4,44 @@ use crate::node_types::{self, EntryKind, Field, NodeTypeMap, Storage, TypeName};
 use crate::trap;
 use std::collections::BTreeMap as Map;
 use std::collections::BTreeSet as Set;
+use std::env;
 use std::path::Path;
 
 use tree_sitter::{Language, Node, Parser, Range, Tree};
 
 pub mod simple;
 
+/// Sets the tracing level based on the environment variables
+/// `RUST_LOG` and `CODEQL_VERBOSITY` (prioritized in that order),
+/// falling back to `warn` if neither is set.
+pub fn set_tracing_level(language: &str) {
+    tracing_subscriber::fmt()
+        .with_target(false)
+        .without_time()
+        .with_level(true)
+        .with_env_filter(
+            tracing_subscriber::EnvFilter::try_from_default_env().unwrap_or_else(
+                |_| -> tracing_subscriber::EnvFilter {
+                    let verbosity = env::var("CODEQL_VERBOSITY")
+                        .map(|v| match v.to_lowercase().as_str() {
+                            "off" | "errors" => "error",
+                            "warnings" => "warn",
+                            "info" | "progress" => "info",
+                            "debug" | "progress+" => "debug",
+                            "trace" | "progress++" | "progress+++" => "trace",
```

Comment:
I'm a bit curious about these levels. They match the ones we [support](https://github.com/github/codeql/blob/619913b55347367b4f5538d4c8728648bfaffb65/python/codeql-extractor.yml#L22) in the Python extractor, but in that case we included things like `debug` and `trace` because these were already existing concepts in the Python extractor. Is this also true of the Rust extractor? If not, it might be more sensible to only support the CodeQL CLI default (but to my mind somewhat opaque) `progress{+,++}` levels.

(Also, mapping `off` to `error` seems a bit odd. What if someone wants to turn errors off?)

Reply:
Good question; I simply took the ones that the C# extractor supports (which matches those supported by the Python extractor).

Regarding `off`, I choose `error` because that appears to be the [lowest supported level](https://docs.rs/tracing-core/latest/tracing_core/struct.Level.html).